src/ast/mod.rs

```diff
@@ -2691,6 +2692,14 @@ pub enum Statement {
         with_options: Vec<SqlOption>,
     },
     /// ```sql
+    /// ALTER TYPE
+    /// See [PostgreSQL](https://www.postgresql.org/docs/current/sql-altertype.html)
+    /// ```
+    AlterType {
+        name: ObjectName,
+        operation: AlterTypeOperation,
+    },
```

Comment:
Can we replace the anonymous enum variants (this and `AlterTypeOperation` with structs? we're slowly [moving a way from using this representation](https://github.com/apache/datafusion-sqlparser-rs/issues/1204)